spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -451,6 +452,8 @@ public StreamLogsController streamLogsController(StreamDeployer streamDeployer)
 			return new StreamLogsController(streamDeployer);
 		}
 
+
+		@Primary
```

Comment:
Yeh I don't think `@Primary` would do what we want here. In fact, it would inject the one marked as primary in both injection targets. 

## Cause
So this is due to the removal of the `LocalVariableTableParameterNameDiscoverer` in SF 6.1 ([details here](https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x#parameter-name-retention). 

## Solution
2 options:

### Option A
Add the `-parameters` flag when we compile.

### Option B
Add `@Qualifier("<yourBeanName>")` to the injection point. Example: 

```
@Bean
@ConditionalOnMissingBean
public AppRegistryController appRegistryController(
		...
		@Qualifier("appRegistryFJPFB") ForkJoinPool appRegistryFJPFB, 
        ...
```

I am in favor of the former option as the latter option will be a Whac-A-Mole.

Reply:
Thank you for the advice!!!!